qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -739,11 +739,15 @@ def _on_proxy_authentication_required(self, url, authenticator,
 
     @pyqtSlot(QUrl, 'QAuthenticator*')
     def _on_authentication_required(self, url, authenticator):
-        # FIXME:qtwebengine support .netrc
-        answer = shared.authentication_required(
-            url, authenticator, abort_on=[self.shutting_down,
-                                          self.load_started])
-        if answer is None:
+        netrc = None
+        if not hasattr(self, 'netrc_used'):
```

Comment:
The reason `hasattr` was used in the old code is because the reply objects were created by Qt, so we couldn't control what attributes is has. Now that we can, there's no reason to use `hasattr`/`setattr`/`delattr` anymore - just set the attribute to `False` initially and test for truthyness.

Also, the attribute should be moved to `TabData` instead. It's a bit unfortunate that it's used only for QtWebEngine, but it's still the best solution, as there can be `QNetworkAccessManager` objects without a tab associated.